gobblin-modules/gobblin-orc/src/main/java/org/apache/gobblin/writer/GobblinBaseOrcWriter.java

```diff
@@ -259,6 +261,15 @@ public void commit()
       throws IOException {
     closeInternal();
     super.commit();
+    if(this.validateORCDuringCommit) {
+      try {
+        OrcFile.createReader(this.outputFile, new OrcFile.ReaderOptions(conf));
+      } catch (IOException ioException) {
+        log.error("Found error when validating ORC file {} during commit phase", this.outputFile, ioException);
+        log.error("Delete the malformed ORC file is successful: {}", this.fs.delete(this.outputFile, false));
```

Comment:
I also notice that the parent class `FsDataWriter` uses HadoopUtils.deletePath.  It's subtle, but perhaps we should use that method instead when deleting because they consider some edge case where it throws an io exception if the file exists but the file fails to delete. I'd imagine if the FileSystem delete covered that edge case, the util method would not have been created, but it's hard to validate when there will be an io exception because it is not documented anywhere.

Reply:
using HadoopUtils.deletePath to delete the file now

Reply:
Still not sure about the retries. If fs delete file fails, we won't delete the file but also won't retry. This works when we call commit because we throw the IO exception to prevent the file from being moved. But we do not do this when we  close the file in the close function, which calls `closeInternal()`. 

If we flush the buffer, we should check after that the file is valid